db/store.go

```diff
@@ -0,0 +1,313 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package db
+
+import (
+	"context"
+
+	"github.com/lens-vm/lens/host-go/config/model"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client"
+)
+
+var _ client.Store = (*store)(nil)
+
+type store struct {
```

Comment:
question: Can you explain why you prefer wrapping the `db` instead of using it directly for the methods bellow?

Reply:
Good point. There's no need for a separate struct anymore. Would you prefer these functions remain in `store.go` or would it be better to move them into `db.go`?

Reply:
`store.go` is fine with me. We could even add documentation at the top saying that the fonction within implement the `client.Store` interface for `db`.